File: trunk/product/src/control/DC0066.php
<?php
/* N21 eBusiness Suite
 * Copyright: Nan21 Electronics srl
 * Generated content.
 * DC0066 Controller: Menu shortcuts
 */

require_once(PATH_CTRL_FRMWK."/Controller.php");

class DC0066 extends Controller {


private function preQuery(&$params, &$where) {
    if (!empty($_REQUEST["QRY_ID"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.ID like :ID";
      $params["ID"] = $_REQUEST["QRY_ID"];
    }
    if (!empty($_REQUEST["QRY_LINK"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.LINK like :LINK";
      $params["LINK"] = $_REQUEST["QRY_LINK"];
    }
    if (!empty($_REQUEST["QRY_MENUITEM_ID"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.MENUITEM_ID like :MENUITEM_ID";
      $params["MENUITEM_ID"] = $_REQUEST["QRY_MENUITEM_ID"];
    }
    if (!empty($_REQUEST["QRY_MENUSHRCT_ID"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.MENUSHRCT_ID like :MENUSHRCT_ID";
      $params["MENUSHRCT_ID"] = $_REQUEST["QRY_MENUSHRCT_ID"];
    }
    if (!empty($_REQUEST["QRY_OWNER"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.OWNER like :OWNER";
      $params["OWNER"] = $_REQUEST["QRY_OWNER"];
    }
    if (!empty($_REQUEST["QRY_POSITION"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.POSITION like :POSITION";
      $params["POSITION"] = $_REQUEST["QRY_POSITION"];
    }
    if (!empty($_REQUEST["QRY_TITLE"])) {
      $where .= (!empty($where))?" and ":"";
      $where .= "ms.TITLE like :TITLE";
      $params["TITLE"] = $_REQUEST["QRY_TITLE"];
    }
}

public function doQuery() {
  try {
    $start = nvl($this->getRequestParam("start"), 0);
    $limit = nvl($this->getRequestParam("limit"),20);
    $orderBy = (!empty($_REQUEST["sort"]))?$_REQUEST["sort"]:"";
    $orderSense = (!empty($_REQUEST["dir"]))?$_REQUEST["dir"]:"";
    $orderByClause = (!empty($orderBy))? "order by $orderBy $orderSense " : "" ;
    $where = "";
    $params = array();
    $this->preQuery($params, $where);
    if (!empty($where)) {
      $where = " where ".$where;
    }
    $sql = "select 
                ms.CREATEDBY
                ,ms.CREATEDON
                ,ms.ID
                ,ms.LINK
                ,ms.MENUITEM_ID
                ,ms.MENUSHRCT_ID
                ,ms.MODIFIEDBY
                ,ms.MODIFIEDON
                ,ms.OWNER
                ,ms.POSITION
                ,ms.TITLE
            from MENU_SHORTCUT ms $where $orderByClause ";
    $this->logger->debug($sql);
    $rs = $this->db->SelectLimit($sql, $limit, $start, $params);
    $rsCount = $this->db->Execute("select count(*) TOTALCOUNT from (".$sql.") t", $params);
    $rsCount->MoveFirst();
    $columns = array(
      "CREATEDBY"
      ,"CREATEDON"
      ,"ID"
      ,"LINK"
      ,"MENUITEM_ID"
      ,"MENUSHRCT_ID"
      ,"MODIFIEDBY"
      ,"MODIFIEDON"
      ,"OWNER"
      ,"POSITION"
      ,"TITLE"
      );
    $dataOut = $this->serializeCursor($rs,$columns, $this->query_data_format);
    if ($this->query_data_format == "xml" ) {header("Content-type: application/xml");}
    print $this->formatQueryResponseData($dataOut, $rsCount->fields["TOTALCOUNT"]);
  }catch(Exception  $e) {
    System::sendActionErrorJson( $e->getMessage());
  }
} /* end function doQuery  */


public function doExport() {
  try {
    $start = nvl($this->getRequestParam("start"), 0);
    $limit = nvl($this->getRequestParam("limit"),20);
    $groupBy = (!empty($_REQUEST["groupBy"]))?$_REQUEST["groupBy"]:"";
    $orderBy = (!empty($_REQUEST["sort"]))?$_REQUEST["sort"]:"";
    $orderSense = (!empty($_REQUEST["dir"]))?$_REQUEST["dir"]:"";
    $orderByClause = (!empty($orderBy))? "order by $orderBy $orderSense " : "" ;
    $where = "";
    $params = array();
    $this->preQuery($params, $where);
    if (!empty($where)) {
      $where = " where ".$where;
    }
    $sql = "select 
                ms.ID
                ,ms.POSITION
                ,ms.OWNER
                ,ms.TITLE
                ,ms.LINK
                ,ms.MENUSHRCT_ID
                ,ms.MENUITEM_ID
                ,ms.CREATEDON
                ,ms.CREATEDBY
                ,ms.MODIFIEDON
                ,ms.MODIFIEDBY
            from MENU_SHORTCUT ms $where $orderByClause ";
    $rs = $this->db->Execute($sql, $params);
    $rsCount = $this->db->Execute("select count(*) TOTALCOUNT from (".$sql.") t", $params);
    $rsCount->MoveFirst();
    $columns = array(
     "ID"
     ,"POSITION"
     ,"OWNER"
     ,"TITLE"
     ,"LINK"
     ,"MENUSHRCT_ID"
     ,"MENUITEM_ID"
     ,"CREATEDON"
     ,"CREATEDBY"
     ,"MODIFIEDON"
     ,"MODIFIEDBY"
      );
    if (!empty($_REQUEST["_p_disp_cols"])) {
      $columns = explode("|",$_REQUEST["_p_disp_cols"]);
    }
    if ($this->getExpFormat() == "csv" ) {
      $dataOut = $this->serializeCursor($rs,$columns,"csv");
    } else {
      $dataOut = $this->serializeCursor($rs,$columns,"xml");
      $dataOut = "<records>".$dataOut."</records>";
      $dataOut = "<queryParams>".$this->serializeArray($params,"xml")."</queryParams>".$dataOut;
      $dataOut = "<columnDef>".$this->columnDefForExport($columns,$this->fieldDef,true).$this->columnDefForExport(array_diff(array_keys($params), $columns),$this->fieldDef,false)."</columnDef>".$dataOut;
      $dataOut = "<staticText>".$this->exportLocalizedStaticText()."</staticText>".$dataOut;
      $dataOut = "<groupBy>".$groupBy."</groupBy>".$dataOut;
      $dataOut = "<reportData  title=\"".$this->getDcTitle()."\" by=\"".$_SESSION["user"]["userName"]."\" on=\"".date(DATE_FORMAT)."\">".$dataOut."</reportData>";
    }
    $this->beginExport();
    print $dataOut;
    $this->endExport();
  }catch(Exception  $e) {
    System::sendActionErrorJson( $e->getMessage());
  }
} /* end function doExport  */


public function fetchRecord() {
  try {
    $RECORD = array();
    $RECORD["ID"] = $this->getRequestParam("ID");
    if (empty($RECORD["ID"])) { throw new Exception("Missing value for primary key field ID in DC0066.fetchRecord().");}
    $pkArray = array("ID" => $RECORD["ID"]);
    $this->findByPk($pkArray, $RECORD);
    print "{success:true, data:".json_encode($RECORD)."}";
  }catch(Exception  $e) {
    System::sendActionErrorJson( $e->getMessage());
  }
} /* end function fetchRecord */


public function doInsert() {
  $this->logger->debug("start: ".$this->dcName.".doInsert");
  try {
    $RECORD = array();
    $RECORD["_p_record_status"] = $this->getRequestParam("_p_record_status");
    $RECORD["_p_store_recId"] = $this->getRequestParam("_p_store_recId");
    $RECORD["CREATEDBY"] = $this->getRequestParam("CREATEDBY");
    $RECORD["CREATEDON"] = $this->getRequestParam("CREATEDON");
    $RECORD["ID"] = $this->getRequestParam("ID");
    $RECORD["LINK"] = $this->getRequestParam("LINK");
    $RECORD["MENUITEM_ID"] = $this->getRequestParam("MENUITEM_ID");
    $RECORD["MENUSHRCT_ID"] = $this->getRequestParam("MENUSHRCT_ID");
    $RECORD["MODIFIEDBY"] = $this->getRequestParam("MODIFIEDBY");
    $RECORD["MODIFIEDON"] = $this->getRequestParam("MODIFIEDON");
    $RECORD["OWNER"] = $this->getRequestParam("OWNER");
    $RECORD["POSITION"] = $this->getRequestParam("POSITION");
    $RECORD["TITLE"] = $this->getRequestParam("TITLE");
    $sql = "insert into MENU_SHORTCUT(
                 ID
                ,LINK
                ,MENUITEM_ID
                ,MENUSHRCT_ID
                ,OWNER
                ,POSITION
                ,TITLE
            ) values ( 
                 :ID
                ,:LINK
                ,:MENUITEM_ID
                ,:MENUSHRCT_ID
                ,:OWNER
                ,:POSITION
                ,:TITLE
    )";
    $stmt = $this->db->prepare($sql);
    $_seq = $this->db->execute("select SEQ_MENUSHRCT_ID.nextval seq_val from dual")->fetchRow();
    $RECORD["ID"] = $_seq["SEQ_VAL"];
    $this->logger->debug("insert of RECORD: ".$this->logger->map2string($RECORD) );
    $this->db->Execute($stmt, $RECORD);
    $pkArray = array("ID" => $RECORD["ID"]);
    $this->findByPk($pkArray, $RECORD);
    print "{success:true, data:".json_encode($RECORD)."}";
    $this->logger->debug("end: ".$this->dcName.".doInsert");
  }catch(Exception  $e) {
    System::sendActionErrorJson( $e->getMessage());
  }
} /* end function doInsert */


public function doUpdate() {
  try {
    $RECORD["_p_record_status"] = $this->getRequestParam("_p_record_status");
    $RECORD["_p_store_recId"] = $this->getRequestParam("_p_store_recId");
    $RECORD["CREATEDBY"] = $this->getRequestParam("CREATEDBY");
    $RECORD["CREATEDON"] = $this->getRequestParam("CREATEDON");
    $RECORD["ID"] = $this->getRequestParam("ID");
    $RECORD["LINK"] = $this->getRequestParam("LINK");
    $RECORD["MENUITEM_ID"] = $this->getRequestParam("MENUITEM_ID");
    $RECORD["MENUSHRCT_ID"] = $this->getRequestParam("MENUSHRCT_ID");
    $RECORD["MODIFIEDBY"] = $this->getRequestParam("MODIFIEDBY");
    $RECORD["MODIFIEDON"] = $this->getRequestParam("MODIFIEDON");
    $RECORD["OWNER"] = $this->getRequestParam("OWNER");
    $RECORD["POSITION"] = $this->getRequestParam("POSITION");
    $RECORD["TITLE"] = $this->getRequestParam("TITLE");
    $sql = "update MENU_SHORTCUT set 
                 CREATEDBY=:CREATEDBY
                ,CREATEDON=:CREATEDON
                ,LINK=:LINK
                ,MENUITEM_ID=:MENUITEM_ID
                ,MENUSHRCT_ID=:MENUSHRCT_ID
                ,MODIFIEDBY=:MODIFIEDBY
                ,MODIFIEDON=:MODIFIEDON
                ,OWNER=:OWNER
                ,POSITION=:POSITION
                ,TITLE=:TITLE
    where 
           ID= :ID
    ";
    $stmt = $this->db->prepare($sql);
    $this->db->Execute($stmt, $RECORD);
    print "{success:true, data:".json_encode($RECORD)."}";
  }catch(Exception  $e) {
    System::sendActionErrorJson( $e->getMessage());
  }
} /* end function doUpdate */


public function doDelete() {
  $this->logger->debug("Start: ".$this->dcName.".doDelete");
  try {
    $RECORD["ID"] = $this->getRequestParam("ID");
    if (empty($RECORD["ID"])) { throw new Exception("Missing value for primary key field ID in DC0066.doDelete().");}
    $sql = "delete from MENU_SHORTCUT where 
           ID= :ID
    ";
    $stmt = $this->db->prepare($sql);
    $this->db->Execute($stmt, $RECORD);
    print "{success:true, data:".json_encode($RECORD)."}";
    $this->logger->debug("End: ".$this->dcName.".doDelete");
  }catch(Exception  $e) {
    System::sendActionErrorJson( $e->getMessage());
  }
} /* end function doDelete */


public function initNewRecord() {
  try {
    $RECORD["CREATEDBY"] = $this->getRequestParam("CREATEDBY");
    $RECORD["CREATEDON"] = $this->getRequestParam("CREATEDON");
    $RECORD["ID"] = $this->getRequestParam("ID");
    $RECORD["LINK"] = $this->getRequestParam("LINK");
    $RECORD["MENUITEM_ID"] = $this->getRequestParam("MENUITEM_ID");
    $RECORD["MENUSHRCT_ID"] = $this->getRequestParam("MENUSHRCT_ID");
    $RECORD["MODIFIEDBY"] = $this->getRequestParam("MODIFIEDBY");
    $RECORD["MODIFIEDON"] = $this->getRequestParam("MODIFIEDON");
    $RECORD["OWNER"] = $this->getRequestParam("OWNER");
    $RECORD["POSITION"] = $this->getRequestParam("POSITION");
    $RECORD["TITLE"] = $this->getRequestParam("TITLE");
    $RECORD["_p_record_status"] = "insert";
    $this->setFieldInitialValues($RECORD, "DC0066");
    print "{success:true, data:".json_encode($RECORD)."}";
  }catch(Exception  $e) {
    System::sendActionErrorJson( $e->getMessage());
  }
} /* end function initNewRecord */

private function findByPk(&$pkCols, &$record) {
    $sql = "select 
                ms.CREATEDBY
                ,ms.CREATEDON
                ,ms.ID
                ,ms.LINK
                ,ms.MENUITEM_ID
                ,ms.MENUSHRCT_ID
                ,ms.MODIFIEDBY
                ,ms.MODIFIEDON
                ,ms.OWNER
                ,ms.POSITION
                ,ms.TITLE
            from MENU_SHORTCUT ms
         where 
           ms.ID= :ID
            ";
    $rs = $this->db->Execute($sql, $pkCols);
    $row = $rs->FetchRow();
    $this->copyArrayValues($record, $row);
} /* end function findByPk  */

private  $fieldDef = array(
  "CREATEDBY" => array("DATA_TYPE" => "STRING")
  ,"CREATEDON" => array("DATA_TYPE" => "DATE")
  ,"ID" => array("DATA_TYPE" => "NUMBER")
  ,"LINK" => array("DATA_TYPE" => "STRING")
  ,"MENUITEM_ID" => array("DATA_TYPE" => "NUMBER")
  ,"MENUSHRCT_ID" => array("DATA_TYPE" => "NUMBER")
  ,"MODIFIEDBY" => array("DATA_TYPE" => "STRING")
  ,"MODIFIEDON" => array("DATA_TYPE" => "DATE")
  ,"OWNER" => array("DATA_TYPE" => "STRING")
  ,"POSITION" => array("DATA_TYPE" => "NUMBER")
  ,"TITLE" => array("DATA_TYPE" => "STRING")
);


private function readRequest(&$RECORD) {
     if (isset($_REQUEST["CREATEDBY"] )) { $RECORD["CREATEDBY"] = $this->getRequestParam("CREATEDBY"); }
     if (isset($_REQUEST["CREATEDON"] )) { $RECORD["CREATEDON"] = $this->getRequestParam("CREATEDON"); }
     if (isset($_REQUEST["ID"] )) { $RECORD["ID"] = $this->getRequestParam("ID"); }
     if (isset($_REQUEST["LINK"] )) { $RECORD["LINK"] = $this->getRequestParam("LINK"); }
     if (isset($_REQUEST["MENUITEM_ID"] )) { $RECORD["MENUITEM_ID"] = $this->getRequestParam("MENUITEM_ID"); }
     if (isset($_REQUEST["MENUSHRCT_ID"] )) { $RECORD["MENUSHRCT_ID"] = $this->getRequestParam("MENUSHRCT_ID"); }
     if (isset($_REQUEST["MODIFIEDBY"] )) { $RECORD["MODIFIEDBY"] = $this->getRequestParam("MODIFIEDBY"); }
     if (isset($_REQUEST["MODIFIEDON"] )) { $RECORD["MODIFIEDON"] = $this->getRequestParam("MODIFIEDON"); }
     if (isset($_REQUEST["OWNER"] )) { $RECORD["OWNER"] = $this->getRequestParam("OWNER"); }
     if (isset($_REQUEST["POSITION"] )) { $RECORD["POSITION"] = $this->getRequestParam("POSITION"); }
     if (isset($_REQUEST["TITLE"] )) { $RECORD["TITLE"] = $this->getRequestParam("TITLE"); }
} /* end function readRequest  */


public function callProcedure($pName) {
  $this->logger->debug("Start: ".$this->dcName.".callProcedure(".$pName.")");
  try {
    $RECORD = array();
    $this->readRequest($RECORD);
    print "{success:true, data:".json_encode($RECORD)."}";
    $this->logger->debug("End: ".$this->dcName.".callProcedure");
  }catch(Exception  $e) {
      System::sendActionErrorJson( $e->getMessage());
  }
} /* end function callProcedure */

}
?>

File: trunk/ServerJava/src/net/nan21/ebs/dc/DC0066.java
/* N21 eBusiness Suite
 * Copyright: Nan21 Electronics srl
 * Generated content.
 * DC0066 DC Controller: Menu shortcuts
 */

package net.nan21.ebs.dc;


import java.sql.Connection;
import java.sql.ResultSet;
import java.util.Date;
import java.util.HashMap;
import java.util.Properties;
import javax.servlet.http.HttpServletResponse;
import net.nan21.ebs.lib.CollectionUtils;
import net.nan21.ebs.lib.AbstractDataControl;
import net.nan21.ebs.lib.FieldDef;
import net.nan21.ebs.lib.HttpRequest;
import net.nan21.ebs.lib.HttpSession;
import net.nan21.ebs.lib.IDataControl;
import net.nan21.ebs.lib.DbManager;

public class DC0066 extends AbstractDataControl implements IDataControl {

  public void init (HttpRequest request, HttpServletResponse response, HttpSession session, DbManager dbm) throws Exception {
    this._initFields();
    super.init(request, response, session, dbm);
  }

private void preQuery() {
    if (this.request.getParam("QRY_ID") != null && !this.request.getParam("QRY_ID").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.ID like :ID");
      this.queryParams.put("ID",(String)this.request.getParam("QRY_ID"));
    }
    if (this.request.getParam("QRY_LINK") != null && !this.request.getParam("QRY_LINK").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.LINK like :LINK");
      this.queryParams.put("LINK",(String)this.request.getParam("QRY_LINK"));
    }
    if (this.request.getParam("QRY_MENUITEM_ID") != null && !this.request.getParam("QRY_MENUITEM_ID").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.MENUITEM_ID like :MENUITEM_ID");
      this.queryParams.put("MENUITEM_ID",(String)this.request.getParam("QRY_MENUITEM_ID"));
    }
    if (this.request.getParam("QRY_MENUSHRCT_ID") != null && !this.request.getParam("QRY_MENUSHRCT_ID").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.MENUSHRCT_ID like :MENUSHRCT_ID");
      this.queryParams.put("MENUSHRCT_ID",(String)this.request.getParam("QRY_MENUSHRCT_ID"));
    }
    if (this.request.getParam("QRY_OWNER") != null && !this.request.getParam("QRY_OWNER").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.OWNER like :OWNER");
      this.queryParams.put("OWNER",(String)this.request.getParam("QRY_OWNER"));
    }
    if (this.request.getParam("QRY_POSITION") != null && !this.request.getParam("QRY_POSITION").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.POSITION like :POSITION");
      this.queryParams.put("POSITION",(String)this.request.getParam("QRY_POSITION"));
    }
    if (this.request.getParam("QRY_TITLE") != null && !this.request.getParam("QRY_TITLE").equals("")) {
      this.queryWhere.append(( this.queryWhere.length() > 0 )?" and ":"");
      this.queryWhere.append("ms.TITLE like :TITLE");
      this.queryParams.put("TITLE",(String)this.request.getParam("QRY_TITLE"));
    }
}

public void doQuery() throws Exception {
    this.prepareQueryContext();
    this.preQuery();
    this.queryWhere.insert(0, (this.queryWhere.length()>0)?" where ":"");
    String sql = "select "+ 
               " ms.CREATEDBY"+
               " ,to_char(ms.CREATEDON,'"+this.DATE_FORMAT_DB+"') CREATEDON"+
               " ,ms.ID"+
               " ,ms.LINK"+
               " ,ms.MENUITEM_ID"+
               " ,ms.MENUSHRCT_ID"+
               " ,ms.MODIFIEDBY"+
               " ,to_char(ms.MODIFIEDON,'"+this.DATE_FORMAT_DB+"') MODIFIEDON"+
               " ,ms.OWNER"+
               " ,ms.POSITION"+
               " ,ms.TITLE"+
           " from MENU_SHORTCUT ms "+this.queryWhere.toString()+" "+this.queryOrderBy;
    this.writeResultDoQuery(sql);
} 


public void doExport() throws Exception {
    this.prepareQueryContext();
    this.preQuery();
    this.queryWhere.insert(0, (this.queryWhere.length()>0)?" where ":"");
    String sql = "select "+ 
               " ms.ID"+
               " ,ms.POSITION"+
               " ,ms.OWNER"+
               " ,ms.TITLE"+
               " ,ms.LINK"+
               " ,ms.MENUSHRCT_ID"+
               " ,ms.MENUITEM_ID"+
               " ,to_char(ms.CREATEDON,'"+this.DATE_FORMAT_DB+"') CREATEDON"+
               " ,ms.CREATEDBY"+
               " ,to_char(ms.MODIFIEDON,'"+this.DATE_FORMAT_DB+"') MODIFIEDON"+
               " ,ms.MODIFIEDBY"+
           " from MENU_SHORTCUT ms "+this.queryWhere.toString()+" "+this.queryOrderBy;
    this.writeResultDoExport(sql);
}

 	public void fetchRecord()  throws Exception  {
     this.populateRecordPkFromRequest();
     this.findByPk();
       this.writeResultFetchRecord();	 
   }

public void doInsert()  throws Exception {
  this.populateRecordFromRequest(); 
  this.populateRecordWithClientSpecific();
    String sql = "insert into MENU_SHORTCUT("+
               "  ID"+
               " ,LINK"+
               " ,MENUITEM_ID"+
               " ,MENUSHRCT_ID"+
               " ,OWNER"+
               " ,POSITION"+
               " ,TITLE"+
           " ) values ( "+
               "  :ID"+
               " ,:LINK"+
               " ,:MENUITEM_ID"+
               " ,:MENUSHRCT_ID"+
               " ,:OWNER"+
               " ,:POSITION"+
               " ,:TITLE"+
    ")";
    this.record.put("ID",   dbm.getSequenceNextValue("SEQ_MENUSHRCT_ID")  );
    dbm.executeStatement(sql, this.record);
    this.populateRecordPkFromRecord();
    this.findByPk();
    this.writeResultDoInsert();
}

public void doUpdate()  throws Exception {
    this.populateRecordFromRequest();
    this.populateRecordWithClientSpecific();
    String sql = "update MENU_SHORTCUT set "+
               "  CREATEDBY=:CREATEDBY"+ 
               " ,CREATEDON=:CREATEDON"+ 
               " ,LINK=:LINK"+ 
               " ,MENUITEM_ID=:MENUITEM_ID"+ 
               " ,MENUSHRCT_ID=:MENUSHRCT_ID"+ 
               " ,MODIFIEDBY=:MODIFIEDBY"+ 
               " ,MODIFIEDON=:MODIFIEDON"+ 
               " ,OWNER=:OWNER"+ 
               " ,POSITION=:POSITION"+ 
               " ,TITLE=:TITLE"+ 
   " where "+ 
     "      ID= :ID"+
    "";
    dbm.executeStatement(sql, this.record);
    this.populateRecordPkFromRecord();
    this.findByPk();
    this.writeResultDoUpdate();	
}

public void doDelete() throws Exception {
    this.populateRecordPkFromRequest();
   String sql = "delete from MENU_SHORTCUT where "+
     "      ID= :ID"+
    "";
    dbm.executeStatement(sql, this.recordPk);
   this.writeResultDoDelete();
}

public void initNewRecord() throws Exception {
    this.populateRecordFromRequest();
    this.record.put("_p_record_status", "insert");
    this.writeResultInitNewRecord();
}

private void findByPk()  throws Exception {
    String sql = "select "+ 
               " ms.CREATEDBY"+
               " ,to_char(ms.CREATEDON,'"+this.DATE_FORMAT_DB+"') CREATEDON"+
               " ,ms.ID"+
               " ,ms.LINK"+
               " ,ms.MENUITEM_ID"+
               " ,ms.MENUSHRCT_ID"+
               " ,ms.MODIFIEDBY"+
               " ,to_char(ms.MODIFIEDON,'"+this.DATE_FORMAT_DB+"') MODIFIEDON"+
               " ,ms.OWNER"+
               " ,ms.POSITION"+
               " ,ms.TITLE"+
           " from MENU_SHORTCUT ms"+
        " where "+
     "      ms.ID= :ID"+ 
          "";
    this.record = dbm.executeQuery(sql, this.recordPk).get(0); 
} 


public void callProcedure(String pName)  throws Exception {
    this.populateRecordFromRequest();
}


	private void  _initFields() {
	  this.fields = new HashMap<String, FieldDef>();
	  this.fields.put("CREATEDBY", new FieldDef("STRING"));
	  this.fields.put("CREATEDON", new FieldDef("DATE"));
	  this.fields.put("ID", new FieldDef("NUMBER"));
	  this.fields.put("LINK", new FieldDef("STRING"));
	  this.fields.put("MENUITEM_ID", new FieldDef("NUMBER"));
	  this.fields.put("MENUSHRCT_ID", new FieldDef("NUMBER"));
	  this.fields.put("MODIFIEDBY", new FieldDef("STRING"));
	  this.fields.put("MODIFIEDON", new FieldDef("DATE"));
	  this.fields.put("OWNER", new FieldDef("STRING"));
	  this.fields.put("POSITION", new FieldDef("NUMBER"));
	  this.fields.put("TITLE", new FieldDef("STRING"));
	  String[] _pkFields = {"ID"};
	  this.pkFields = _pkFields;
	}

public void doCustomAction(String action) {}
}
